### DISCUSSION TITLE
It there a general way to add the border to every floated windows? (lazy.nvim, mason, lazy terminal etc.)

### DISCUSSION THREAD

**Discussion by neuromaancer** [LazyVim/LazyVim#2092] [Q&A] [answered]
I think it mixes up sometime my code and these windows together.

**Comment by dpetka2001** [comment, marked as answer]
For `lazy.nvim` related floating windows you could add in `/lua/config/lazy.lua` inside the `require("lazy").setup` the following
```lua
  ui = {
    border = "rounded",
  },
```
and for mason in `/lua/plugins/mason.lua`
```lua
return {
  {
    "williamboman/mason.nvim",
    opts = {
      ui = {
        border = "rounded",
      },
    },
  },
}
```

**Reply by neuromaancer** [reply]
Thanks, what about the lazy terminal?

**Reply by neuromaancer** [reply]
Oh, actually it works for terminals as well! Thanks!

**Reply by nounder** [reply]
Any idea why changing options for mason applies to floating terminal?

**Reply by dotfrag** [reply]
I believe changing mason opts doesn't affect the floating terminal, it's the lazy option that does it, because the terminal is created using lazy:

```lua
require("lazy.util").float_term(cmd, opts)
```

**Comment by nounder** [comment]
mason option change didn't worked for me. An alternative solution is to put following in `keymaps.lua`:

```
local Util = require("lazyvim.util")

vim.keymap.set("n", "<C-/>", function()
  Util.terminal(nil, { border = "rounded" })
end, { desc = "Terminal (root dir )" })
```

**Comment by zaynOm** [comment]
any idea on how can the same result be achieved for the hover popup

**Reply by dotfrag** [reply]
That is done with `noice`. The related option is:

```lua
opts = {
  presets = {
    lsp_doc_border = true, -- add a border to hover docs and signature help
  }
}
```

**Reply by zaynOm** [reply]
This works perfectly, the only issue I have now is the LSP completion popup does not have the same colorscheme instead it has a black bg

**Reply by dotfrag** [reply]
Can you post a screenshot? Mine looks the same with border on and off.

**Reply by dpetka2001** [reply]
You don't need the `return opts` except if you want to override all the default values and return your own custom ones. Just use a table like @dotfrag already showed or change directly the value with `opts.presets["whatever_you_want"]` or even 
```lua
opts.presets = {
  lsp_doc_border = true
}
```
All are equivalent.

PS: my reply was towards @iamxiaojianzheng but apparently deleted his comment when I was replying.

**Reply by iamxiaojianzheng** [reply]
@dpetka2001  hhhh, Sorry, I was aware of the problem not long after I posted the comment.

**Reply by iamxiaojianzheng** [reply]
@dpetka2001 

Can you take a look at this problem for me?

https://github.com/LazyVim/LazyVim/discussions/3688

**Reply by dpetka2001** [reply]
Oh sorry but unfortunately I don't know how to do this. `lazyterm` uses default Neovim terminal and that's not to be used for interactive commands as far as I know. You can print something like `ls` or some other simple command but not something interactive. If you want something interactive you should look into [toggleterm](https://github.com/akinsho/toggleterm.nvim) for more advanced features (although I'm not sure if it supports what you want cuz I've never tried it).

**Reply by zaynOm** [reply]
> You don't need the `return opts` except if you want to override all the default values and return your own custom ones. Just use a table like @dotfrag already showed or change directly the value with `opts.presets["whatever_you_want"]` or even
> 
> ```lua
> opts.presets = {
>   lsp_doc_border = true
> }
> ```
> 
> All are equivalent.
> 
> PS: my reply was towards @iamxiaojianzheng but apparently deleted his comment when I was replying.

but in the [docs](https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-customizing-plugin-specs) it says that if you return `opts` it will be merged with the default one

**Reply by dpetka2001** [reply]
You can do 
```lua
opts = function(_, opts)
  opts.presets = {
    -- whatever you want here
  }
end,
```
and you can do 
```lua
opts = function()
  local opts = {
    -- whatever you want here
  }

  return opts
end,
```
The first one will just change the specific values of the default settings. The latter will completely override all default options with your own custom ones.

**Reply by dpetka2001** [reply]
See corresponding section in the [LazyVim starter example.lua](https://github.com/LazyVim/starter/blob/cb79b0e6a9d0ec81041150dc87fe47352a54a2ba/lua/plugins/example.lua#L155-L173).

**Reply by zaynOm** [reply]
> Can you post a screenshot? Mine looks the same with border on and off.

- **Issue**
![image](https://github.com/LazyVim/LazyVim/assets/63248283/7c90b020-a62f-4886-93c3-03da3aa30835)
As you can see the completion & documentation floating windows does not have the transparent background specified in the colorscheme

- **Solution**
![image](https://github.com/LazyVim/LazyVim/assets/63248283/3fb4688c-c590-45b2-8651-7ed454dd981e)
```lua
return {
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      opts.window = {
        completion = {
          border = "rounded",
          winhighlight = "Normal:MyHighlight",
          winblend = 0,
        },
        documentation = {
          border = "rounded",
          winhighlight = "Normal:MyHighlight",
          winblend = 0,
        },
      }
    end,
  },
```

**Reply by dotfrag** [reply]
This could be an issue comig from transparency, sorry I'm not using it to help you test.

**Reply by zaynOm** [reply]
> You can do
> 
> ```lua
> opts = function(_, opts)
>   opts.presets = {
>     -- whatever you want here
>   }
> end,
> ```
> 
> and you can do
> 
> ```lua
> opts = function()
>   local opts = {
>     -- whatever you want here
>   }
> 
>   return opts
> end,
> ```
> 
> The first one will just change the specific values of the default settings. The latter will completely override all default options with your own custom ones.

Yeah you're right, but I'm talking about returning an opts variable not a function.
[like this](https://github.com/LazyVim/starter/blob/cb79b0e6a9d0ec81041150dc87fe47352a54a2ba/lua/plugins/example.lua#L23-L28)

**Reply by dpetka2001** [reply]
It's an issue with the colorscheme he uses which highlight groups it is setting when the theme is transparent and it conflicts with the default nvim-cmp highlight groups probably.

**Reply by dpetka2001** [reply]
@zaynOm My original comment that you quoted was towards @iamxiaojianzheng comment which he then deleted. In his original comment he was using `opts` as a function. The way you used `opts` has no trouble at all merging with the defaults.

> PS: my reply was towards @iamxiaojianzheng but apparently deleted his comment when I was replying.

**Reply by zaynOm** [reply]
@dpetka2001 @dotfrag thx for your time, I was just making sure I'm not using it the wrong way.
can you take a look at this #3685
